src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -271,26 +280,119 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(B
             conversionDiagnostics?.Free();
 
             var intType = GetSpecialType(SpecialType.System_Int32, diagnostics, unconvertedInterpolatedString.Syntax);
-            var arguments = ImmutableArray.Create<BoundExpression>(
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(baseStringLength), intType) { WasCompilerGenerated = true },
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(numFormatHoles), intType) { WasCompilerGenerated = true });
+            int constructorArgumentLength = 2 + (additionalConstructorArguments.IsDefault ? 0 : additionalConstructorArguments.Length);
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(constructorArgumentLength);
+            populateArguments(unconvertedInterpolatedString, additionalConstructorArguments, baseStringLength, numFormatHoles, intType, argumentsBuilder);
+
+            var refKindsBuilder = ArrayBuilder<RefKind>.GetInstance(constructorArgumentLength);
+            refKindsBuilder.Add(RefKind.None);
+            refKindsBuilder.Add(RefKind.None);
+
+            if (!refKinds.IsDefault)
+            {
+                refKindsBuilder.AddRange(refKinds);
+            }
+
+            var nonOutConstructorDiagnostics = BindingDiagnosticBag.GetInstance(template: diagnostics);
+            BoundExpression constructorCall;
+            BoundExpression nonOutConstructorCall = MakeConstructorInvocation(interpolatedStringHandlerType, argumentsBuilder, refKindsBuilder, unconvertedInterpolatedString.Syntax, nonOutConstructorDiagnostics);
+            if (nonOutConstructorCall is not BoundObjectCreationExpression { ResultKind: LookupResultKind.Viable })
```

Comment:
>nonOutConstructorCall is not BoundObjectCreationExpression { ResultKind: LookupResultKind.Viable } [](http://example.com/codeflow?start=16&length=98)

This approach feels somewhat fragile because we have no idea what went wrong and whether we would prefer reporting the errors rather than trying to bind with an additional out argument. For example, what if the only reason the attempt failed is a missing assembly reference? Should we instead ask to add the reference, or move to the second form? I assume there could be other similar situations. Perhaps we should pay attention to the reasons why “best” among “bad” candidates were rejected. I think we mark candidates with the failure kind during overload resolution.

Reply:
I think it should be fine to follow up on the error detection/recovery separately. A PROTOTYPE comment should be good enough for this PR.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/54396](https://github.com/dotnet/roslyn/issues/54396) to follow up.